repo/splitter/splitter_test.go

```diff
@@ -75,66 +50,153 @@ func TestSplitterStability(t *testing.T) {
 		{Pooled(newRabinKarp64SplitterFactory(65536)), 53, 94339, 35875, 131072},
 	}
 
+	getSplitPointsFunctions := map[string]func(data []byte, s Splitter) (minSplit, maxSplit, count int){
+		"getSplitPoints":             getSplitPoints,
+		"getSplitPointsByteByByte":   getSplitPointsByteByByte,
+		"getSplitPointsRandomSlices": getSplitPointsRandomSlices,
+	}
+
 	for _, tc := range cases {
 		tc := tc
 		t.Run(fmt.Sprintf("%v", tc), func(t *testing.T) {
 			t.Parallel()
-			// run each test twice to rule out the possibility of some state leaking through splitter reuse
-			for repeat := 0; repeat < 2; repeat++ {
-				s := tc.factory()
-
-				lastSplit := -1
-				maxSplit := 0
-				minSplit := int(math.MaxInt32)
-				count := 0
-
-				if got, want := s.MaxSegmentSize(), tc.maxSplit; got != want {
-					t.Errorf("unexpected max segment size: %v, want %v", got, want)
-				}
-
-				for i, p := range rnd {
-					if !s.ShouldSplit(p) {
-						continue
-					}
 
-					l := i - lastSplit
-					if l >= maxSplit {
-						maxSplit = l
-					}
+			for name, getSplitPointsFunc := range getSplitPointsFunctions {
```

Comment:
It'd be good to also have the prior implementations (only in tests) and use them as reference, even if they have a different interface.

Reply:
i feel pretty confident that the current implementation has not regressed because of the baseline tests that asserts the distribution of split points, also the fact that outputs of benchmark are exactly the same.

not sure how having two implementations would even work.